packages/clerk-js/src/core/resources/SignIn.ts

```diff
@@ -93,13 +93,10 @@ export class SignIn extends BaseResource implements SignInResource {
         break;
       case 'reset_password_code':
         if (factor.emailAddressId) {
-          config = { emailAddressId: factor.emailAddressId } as ResetPasswordCodeFactorConfig;
-        }
-        if (factor.phoneNumberId) {
-          config = {
-            phoneNumberId: factor.phoneNumberId,
-          } as ResetPasswordCodeFactorConfig;
+          config = { emailAddressId: factor?.emailAddressId } as ResetPasswordCodeFactorConfig;
+          break;
         }
+        config = { phoneNumberId: factor?.phoneNumberId } as ResetPasswordCodeFactorConfig;
```

Comment:
⛔ Let's keep the check for `factor.phoneNumberId` to be safe and maybe throw an error on the other case if the factor is not an email or phone number. Otherwise, we will always use phone that is not guaranteed that it exists.

Reply:
One way to throw the error is [this](https://github.com/clerkinc/javascript/pull/1079#pullrequestreview-1402495153)

Reply:
@SokratisVidros  I'll take another look, but the link you added seems to link back to this PR